File: CPP/00_Pruebas/randomDnd.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <limits>
using namespace std;

int randomNumber(int min, int max) {
    return rand() % (max - min + 1) + min;
}

string razas[] = {"Humano", "Elfo", "Enano", "Mediano", "Dragonborn", "Gnomo", "Medusa", "Tiefling", "Orco"};
string clases[] = {"Barbaro", "Bardo", "Clerigo", "Druida", "Hechicero", "Mago", "Monje", "Paladin", "Picaro", "Explorador", "Guerrero"};
string trasfondos[] = {"Acolito", "Charlatan", "Criminal", "Erudito", "Heroe de guerra", "Noble", "Saber popular", "Sacerdote", "Vagabundo"};
string profesiones[] = {"Alquimista", "Cazador", "Forjador", "Granjero", "Mercenario", "Pescador", "Sastre", "Tavernero", "Médico", "Minero"};

int main() {
    srand(time(0));
    char respuesta;

    do {
        int razaIndex = randomNumber(0, sizeof(razas) / sizeof(razas[0]) - 1);
        int claseIndex = randomNumber(0, sizeof(clases) / sizeof(clases[0]) - 1);
        int trasfondoIndex = randomNumber(0, sizeof(trasfondos) / sizeof(trasfondos[0]) - 1);
        int profesionIndex = randomNumber(0, sizeof(profesiones) / sizeof(profesiones[0]) - 1);

        cout << "Raza: " << razas[razaIndex] << endl;
        cout << "Clase: " << clases[claseIndex] << endl;
        cout << "Trasfondo: " << trasfondos[trasfondoIndex] << endl;
        cout << "Profesion: " << profesiones[profesionIndex] << endl;

        cout << "\n_______________________________________\n";
        cout << "¿Quieres generar otro personaje? (s/n): ";
        cin >> respuesta;
        
        // Limpiar cualquier error en el flujo de entrada
        cin.clear();
        system("cls");

        // Descartar caracteres restantes en el buffer hasta encontrar un salto de línea
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    } while (respuesta == 's' || respuesta == 'S');

    return 0;
}

File: HTML/01_randomDnd/randomDnd.js
const razas = ["Humano", "Elfo", "Enano", "Mediano", "Dragonborn", "Gnomo", "Medusa", "Tiefling", "Orco"];
        const clases = ["Bárbaro", "Bardo", "Clérigo", "Druida", "Hechicero", "Mago", "Monje", "Paladín", "Pícaro", "Explorador", "Guerrero"];
        const trasfondos = ["Acolito", "Charlatán", "Criminal", "Erudito", "Héroe de guerra", "Noble", "Saber popular", "Sacerdote", "Vagabundo"];
        const profesiones = ["Alquimista", "Cazador", "Forjador", "Granjero", "Mercenario", "Pescador", "Sastre", "Tavernero", "Médico", "Minero"];
        const alineamientos = ["Legal Bueno", "Neutral Bueno", "Caótico Bueno", "Legal Neutral", "Neutral", "Caótico Neutral", "Legal Malvado", "Neutral Malvado", "Caótico Malvado"];

        function randomNumber(min, max) {
            return Math.floor(Math.random() * (max - min + 1)) + min;
        }

        function generarPersonaje() {
            const razaIndex = randomNumber(0, razas.length - 1);
            const claseIndex = randomNumber(0, clases.length - 1);
            const trasfondoIndex = randomNumber(0, trasfondos.length - 1);
            const profesionIndex = randomNumber(0, profesiones.length - 1);
            const alineamientoIndex = randomNumber(0, alineamientos.length - 1);

            const output = document.getElementById("output");
            output.innerHTML = `
                <h3>${razas[razaIndex]}</h3>
                <h3>${clases[claseIndex]}</h3>
                <h3>${trasfondos[trasfondoIndex]}</h3>
                <h3>${profesiones[profesionIndex]}</h3>
                <h3>${alineamientos[alineamientoIndex]}</h3>
            `;
        }